<a href="https://colab.research.google.com/github/UnplugCharger/DIGITAL-APRENTICIPPROGRAMME/blob/main/digital_apprentiship_programm.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np


In [2]:
url= 'https://raw.githubusercontent.com/onaio/ona-tech/master/data/water_points.json'
df=pd.read_json(url)
df.head()

water_pay respondent  ... simserial subscriberid
0        no  community  ...       NaN          NaN
1        no  community  ...       NaN          NaN
2        no  community  ...       NaN          NaN
3        no  community  ...       NaN          NaN
4        no  community  ...       NaN          NaN

[5 rows x 48 columns]

In [5]:
df.columns

Index(['water_pay', 'respondent', 'research_asst_name', 'water_used_season',
       '_bamboo_dataset_id', '_deleted_at', 'water_point_condition',
       '_xform_id_string', 'other_point_1km', '_attachments',
       'communities_villages', 'end', 'animal_number', 'water_point_id',
       'start', 'water_connected', 'water_manager_name', '_status',
       'enum_id_1', 'water_lift_mechanism', 'districts_divisions', '_uuid',
       'grid', 'date', 'formhub/uuid', 'road_available', 'water_functioning',
       '_submission_time', 'signal', 'water_source_type', '_geolocation',
       'water_point_image', 'water_point_geocode', 'deviceid',
       'locations_wards', 'water_manager', 'water_developer', '_id',
       'animal_point', 'water_mechanism_plate', 'water_lift_mechanism_type',
       'road_type', 'water_mechanism_plate_units', 'water_mechanism_plate_no',
       'water_not_functioning', 'water_source_type_other', 'simserial',
       'subscriberid'],
      dtype='object')

# The number of water points that are functional, 

In [6]:
df['water_point_condition','water_point_id'].value_counts()

functioning           590
broken                 40
abandoned              36
newly_constructed      28
under_construction     16
na_dn                   2
Name: water_point_condition, dtype: int64

590 water points are functional

There are 623 water points that are functional.

# The number of water points per community

In [4]:
df['communities_villages'].unique()

array(['Gumaryili', 'Selinvoya', 'Nabulugu', 'Nayoku', 'Zangu-Vuga',
       'Arigu', 'Gbaarigu', 'Kurugu', 'Namgurima', 'Zukpeni', 'Tantala',
       'Loagri_1_', 'Kpatarigu', 'Kulbugu', 'Kubore', 'Zanwara', 'Garigu',
       'Gbima', 'Soo', 'Zua', 'Dibisi', 'Alavanyo', 'Abanyeri',
       'Chondema', 'Jaata', 'Nawaasa', 'Fiisa', 'Kanwaasa', 'Jiriwiensa',
       'Kaasa', 'Guuta-Nasa', 'Guuta', 'Zundem', 'Zogsa', 'Longsa',
       'Dorinsa', 'Bechinsa', 'Akpari-yeri', 'Badomsa', 'Zuedema',
       'Tankangsa', 'Banyangsa', 'Tuisa', 'Kpikpaluk', 'Mwalorinsa',
       'Chansa', 'Chanpolinsa', 'Kanbangsa', 'Bandem', 'Jagsa',
       'Logvasgsa', 'Kalaasa', 'Nyandema', 'Luisa', 'Vundema',
       'Jiningsa-Yipaala', 'Piisa', 'Sikabsa', 'Nyankpiensa', 'Jiningsa',
       'Kom', 'Kunkwah', 'Gaadem', 'Suik', 'Jiniensa'], dtype=object)

In [21]:
df2= df[['communities_villages','water_point_id']]
df2

communities_villages water_point_id
0              Gumaryili            xxx
1              Selinvoya            xxx
2              Selinvoya            xxx
3              Selinvoya            xxx
4              Selinvoya            xxx
..                   ...            ...
707                 Suik        411c-24
708              Vundema            Xxx
709              Vundema            Xxx
710             Jiniensa            xxx
711                Jagsa            XXX

[712 rows x 2 columns]

In [31]:
df3=df2.groupby(['communities_villages'])['water_point_id'].count()

In [32]:
df3

communities_villages
Abanyeri        4
Akpari-yeri     3
Alavanyo        3
Arigu          12
Badomsa        27
               ..
Zogsa           6
Zua            28
Zuedema        18
Zukpeni         6
Zundem         30
Name: water_point_id, Length: 65, dtype: int64

# The rank for each community by the percentage of broken water points.

In [34]:
df4=df[df['water_functioning']=='broken']

In [36]:
df4

Empty DataFrame
Columns: [water_pay, respondent, research_asst_name, water_used_season, _bamboo_dataset_id, _deleted_at, water_point_condition, _xform_id_string, other_point_1km, _attachments, communities_villages, end, animal_number, water_point_id, start, water_connected, water_manager_name, _status, enum_id_1, water_lift_mechanism, districts_divisions, _uuid, grid, date, formhub/uuid, road_available, water_functioning, _submission_time, signal, water_source_type, _geolocation, water_point_image, water_point_geocode, deviceid, locations_wards, water_manager, water_developer, _id, animal_point, water_mechanism_plate, water_lift_mechanism_type, road_type, water_mechanism_plate_units, water_mechanism_plate_no, water_not_functioning, water_source_type_other, simserial, subscriberid]
Index: []